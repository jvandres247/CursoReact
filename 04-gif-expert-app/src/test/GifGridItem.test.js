import { shallow } from "enzyme";
import { GifGridItem } from "../components/GifGridItem";
import React from "react";
import "../setupTests";

describe("Pruebas en <GifGridItem/>", () => {
  const title = "Titulo Test";
  const url = "https://localhost.com/imagen.png";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe mostrar <GifGridItem /> correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de tener un parrafo con el titulo', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('Debe de tener la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('Debe tener animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
