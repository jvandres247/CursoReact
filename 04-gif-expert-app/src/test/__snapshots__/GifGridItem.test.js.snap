// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar <GifGridItem /> correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo Test"
    src="https://localhost.com/imagen.png"
  />
  <p>
    Titulo Test
  </p>
</div>
`;
